lib/IRGen/GenClangDecl.cpp

```diff
@@ -82,6 +84,100 @@ clang::Decl *getDeclWithExecutableCode(clang::Decl *decl) {
   return nullptr;
 }
 
+auto walkCallGraphFromCtor(const clang::CXXConstructorDecl *toplevelCtor,
+                           std::set<const clang::Decl *> visited,
+                           std::vector<const clang::Decl *> &FDS) {
+  std::vector<clang::Stmt *> stack;
+
+  // Keep track of decls we have already walked over so that we don't re-walk
+  // over them redundantly.
+  auto visitDecl = [&visited](const clang::Decl *decl) {
+    const bool hasVisited = visited.find(decl) != visited.end();
+    if (!hasVisited) {
+      visited.insert(decl);
+    }
+    return hasVisited;
+  };
+
+  auto HandleCtor = [&stack, &FDS](const clang::CXXConstructorDecl *ctor) {
+    FDS.push_back(ctor);
+    stack.push_back(ctor->getBody());
+    for (auto *init : ctor->inits()) {
+      stack.push_back(init->getInit());
+      if (clang::FieldDecl *field = init->getMember())
+        stack.push_back(field->getInClassInitializer());
+    }
+  };
+
+  auto HandleFunctionDecl = [&stack, &FDS](clang::FunctionDecl *F) {
+    LLVM_DEBUG({
+      llvm::dbgs() << "HANDLE FUNCTION DECL:\n";
+      llvm::dbgs() << "IS INLINE " << (F->isInlineSpecified() ? "YES" : "NO")
+                   << "\n";
+      llvm::dbgs() << "IS TEMPLATE INST "
+                   << (F->isTemplateInstantiation() ? "YES" : "NO") << "\n";
+    });
+
+    if (F->isInlineSpecified() || F->isTemplateInstantiation())
```

Comment:
I'm sure there's an `isInline` method, no?

Reply:
I think `FunctionDecl::isInlined` is what you're looking for but I'm not 100% sure that will cover templates too.

Reply:
I am a little confused, theres IsInline and IsInlineSpecified. I gotta look into which means what.

Reply:
Ahh this is interesting:

```
 /// Determine whether the "inline" keyword was specified for this
  /// function.
  bool isInlineSpecified() const { return FunctionDeclBits.IsInlineSpecified; }

  ///...

  /// Determine whether this function should be inlined, because it is
  /// either marked "inline" or "constexpr" or is a member function of a class
  /// that was defined in the class body.
  bool isInlined() const { return FunctionDeclBits.IsInline; }
```

I think you are right. I want isInlined because that will capture the constexpr case as well. I think template I need a separate one.